contracts/contracts-registry/AbstractContractsRegistry.sol

```diff
@@ -83,122 +83,124 @@ abstract contract AbstractContractsRegistry is Initializable {
 
     /**
      *  @notice The function that returns an implementation of the given proxy contract
-     *  @param name the name of the contract
+     *  @param name_ the name of the contract
      *  @return the implementation address
      */
-    function getImplementation(string memory name) public view returns (address) {
-        address contractProxy = _contracts[name];
+    function getImplementation(string memory name_) public view returns (address) {
+        address contractProxy_ = _contracts[name_];
 
-        require(contractProxy != address(0), "ContractsRegistry: This mapping doesn't exist");
-        require(_isProxy[contractProxy], "ContractsRegistry: Not a proxy contract");
+        require(contractProxy_ != address(0), "ContractsRegistry: this mapping doesn't exist");
+        require(_isProxy[contractProxy_], "ContractsRegistry: not a proxy contract");
 
-        return _proxyUpgrader.getImplementation(contractProxy);
+        return _proxyUpgrader.getImplementation(contractProxy_);
     }
 
     /**
      *  @notice The function that injects the dependencies into the given contract
-     *  @param name the name of the contract
+     *  @param name_ the name of the contract
      */
-    function _injectDependencies(string memory name) internal {
-        address contractAddress = _contracts[name];
+    function _injectDependencies(string memory name_) internal virtual {
```

Comment:
Maybe add a function that takes an extra date for the `setDependencies` function?